### PR TITLE
[1221] Handle saving of AP in support Flow

### PR DESCRIPTION
**Pull request by darokel**
### Context

https://trello.com/c/zyz47TB6/1221-support-add-edit-accrediting-provider-step-3-check-your-answers

### Changes proposed in this pull request

- Implements saving of accredited providers in the Add AP flow for Support
- Sends a new email setup in Notify to associated users
- Cleans up duplicated code from parallel providers work

### Guidance to review

- Go to support console, and find a lead school
- Search and add an accredited provider

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by darokel** [comment]
[Support: Add & Edit Accrediting Provider - Step 3 (Check your answers)](https://trello.com/c/zyz47TB6/1221-support-add-edit-accrediting-provider-step-3-check-your-answers)

**Review by avinhurry — COMMENTED** [review]
Just noticed if I go through the flow on the review app, get to the confirm page, then click change and choose a new accredited provider, it doesn't update on the confirm page 🤔

**Review by goodviber — APPROVED** [review]
lgtm

**Review by avinhurry — APPROVED** [review]
Left a couple of questions/suggestions, but looks good 👍🏼

**Review by victoriawinlow — APPROVED** [review]

File: spec/forms/support/accredited_provider_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        it 'updates the provider with the new accredited provider information' do
+          subject.save!
+
+          expect(model.accrediting_provider_enrichments.count).to eq(2)
+          expect(model.accrediting_provider_enrichments.last.UcasProviderCode).to eq(accredited_provider.provider_code)
+          expect(model.accrediting_provider_enrichments.last.Description).to eq('Foo')
+        end
+      end
+
+      it 'sends a notification email to the accredited provider users' do
```
[the remark this answers is not in the document]
Comment (darokel) [reply spec/forms/support/accredited_provider_form_spec.rb]: nah it's a private method, i've tested the behaviour via the main public method which in this case is `#save`

**Pull request merged**